K8SSAND-1251 ⁃ testInPlaceRestore integration test can fail with a panic from nil pointer dereference

<!--

Feel free to ask questions on Discord at https://discord.com/invite/y4s64xC9 or in the forum at https://forum.k8ssandra.io/. 
-->

**What happened?**
In this test [run](https://github.com/k8ssandra/k8ssandra-operator/runs/5203253688?check_suite_focus=true) the testInPlaceRestore integration test failed with the following panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x19a0bb6]

goroutine 5588 [running]:
github.com/k8ssandra/k8ssandra-operator/controllers/medusa.testInPlaceRestore.func2(0x0)
	/home/runner/work/k8ssandra-operator/k8ssandra-operator/controllers/medusa/restore_controller_test.go:108 +0x2f6
github.com/stretchr/testify/assert.Eventually.func1(0xc007ca21c0, 0xc007c5b170)
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.7.0/assert/assertions.go:1655 +0x27
created by github.com/stretchr/testify/assert.Eventually
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.7.0/assert/assertions.go:1655 +0x22a
FAIL	github.com/k8ssandra/k8ssandra-operator/controllers/medusa	122.818s
ok  	github.com/k8ssandra/k8ssandra-operator/controllers/reaper	32.838s	coverage: 65.7% of statements
ok  	github.com/k8ssandra/k8ssandra-operator/controllers/replication	103.047s	coverage: 71.8% of statements
ok  	github.com/k8ssandra/k8ssandra-operator/controllers/stargate	60.272s	coverage: 56.6% of statements
FAIL
make: *** [Makefile:125: test] Error 1
Error: Process completed with exit code 2.
```
Looks like this is caused by an incorrect nil check on a status condition.

**Did you expect to see something different?**
The test should not generate this error.

**How to reproduce it (as minimally and precisely as possible)**:
I have run `make test` multiple times locally and have not reproduced. It's easy to see what change is needed from manual code inspection.

**Environment**

* K8ssandra Operator version:

    `v0.5.0`

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-1251) by [Unito](https://www.unito.io)
┆friendlyId: K8SSAND-1251
┆priority: Medium

--- a/CHANGELOG/CHANGELOG-1.0.md
+++ b/CHANGELOG/CHANGELOG-1.0.md
@@ -17,6 +17,7 @@ When cutting a new release, update the `unreleased` heading to the tag being gen
 
 * [BUGFIX] [#381](https://github.com/k8ssandra/k8ssandra-operator/issues/381) Add additionalSeeds property to support migrations
 * [TESTING] [#332](https://github.com/k8ssandra/k8ssandra-operator/issues/332) Configure mocks to avoid panics
+* [TESTING] [#407](https://github.com/k8ssandra/k8ssandra-operator/issues/407) Fix `nil` condition checks in backup/restore integration tests
 
 ## v0.5.0 2022-02-10
 

--- a/controllers/medusa/backup_controller_test.go
+++ b/controllers/medusa/backup_controller_test.go
@@ -115,7 +115,7 @@ func testBackupDatacenter(t *testing.T, ctx context.Context, f *framework.Framew
 		}
 
 		condition := findDatacenterCondition(k8ssandraStatus.Cassandra, cassdcapi.DatacenterScalingUp)
-		return !(condition == nil && condition.Status == corev1.ConditionFalse)
+		return condition != nil && condition.Status == corev1.ConditionTrue
 	}, timeout, interval, "timed out waiting for K8ssandraCluster status update")
 
 	dc1 := &cassdcapi.CassandraDatacenter{}

--- a/controllers/medusa/restore_controller_test.go
+++ b/controllers/medusa/restore_controller_test.go
@@ -105,7 +105,7 @@ func testInPlaceRestore(t *testing.T, ctx context.Context, f *framework.Framewor
 		}
 
 		condition := findDatacenterCondition(k8ssandraStatus.Cassandra, cassdcapi.DatacenterScalingUp)
-		return !(condition == nil && condition.Status == corev1.ConditionFalse)
+		return condition != nil && condition.Status == corev1.ConditionTrue
 	}, timeout, interval, "timed out waiting for K8ssandraCluster status update")
 
 	dc1 := &cassdcapi.CassandraDatacenter{}

